runtime/common/src/parachains.rs

```diff
@@ -781,6 +783,14 @@ impl<T: Trait> Module<T> {
 			let validator_group = validator_groups.group_for(para_id)
 				.ok_or(Error::<T>::NoValidatorGroup)?;
 
+			let actual_head = Self::parachain_head(&para_id)
+				.map(primitives::parachain::HeadData);
+
+			ensure!(
+				actual_head.as_ref() == Some(&candidate.candidate.parent_head),
```

Comment:
Do we really want to skip the inclusion of all parachain heads, when one is not correct?

(Will the validator that gave us this be slashed later?)

(I see that we do the same below for other checks as well, but the same question still holds)

Reply:
Yes - this gives us the property that every included head in the parachain block passes checks. These checks are relatively cheap, so it's fine to do them.